crates/core/agent_mapper_interface/src/topic.rs

```diff
@@ -1,17 +1,22 @@
-use crate::sm_c8y_mapper::error::MapperTopicError;
 use mqtt_client::{MqttClientError, Topic};
 use std::convert::{TryFrom, TryInto};
 
+#[derive(thiserror::Error, Debug)]
+pub enum TopicError {
+    #[error("Topic {topic} is unknown.")]
+    UnknownTopic { topic: String },
+}
+
 #[derive(Debug, Clone, PartialEq)]
-pub(crate) enum IncomingTopic {
+pub enum IncomingTopic {
     SoftwareListResponse,
     SoftwareUpdateResponse,
     SmartRestRequest,
```

Comment:
The `SmartRestRequest` topic is unrelated to the agent and must be removed from here.

Reply:
separated the agent_interface topics and c8y topics.